### PR TITLE
Update gradle plugins, circle and fix lint errors

### PR DESCRIPTION
**Pull request by glynternet**
Lots of the gradle build and packages were outdated.
I wanted to update all of these before making some changes to the repo.

To note:
* I copied `.baseline/checkstyle/checkstyle.xml` and `.baseline/checkstyle/checkstyle-suppressions.xml` from `palantir/conjure-java`
* Baseline plugin usage copied from `conjure-java`
* I had to run spotlessApply which has produced much of the changes
* The test case expected values now have alphabetically-sorted JSON fields
* LinkedList in TemplateTypeVisitor.java replaced with ArrayDeque
* Root project name changed to 'palantir-conjure-postman' to avoid name clash with subproject conjure-postman. This matches the pattern from conjure-java

### CODE REVIEW

File: .baseline/checkstyle/checkstyle-suppressions.xml [hunk 1]
```diff
@@ -21,7 +25,5 @@
     <suppress files="\.(bdr|eot|gif|gzip|jar|json|otf|png|svg|ttf|woff|zip)$" checks="NewlineAtEndOfFile" />
 
     <!-- Generated code should not be subjected to checkstyle. -->
-    <suppress files="[/\\].*[/\\]generated_src[/\\]" checks="." />
-    <suppress files="[/\\].*[/\\]generated_testSrc[/\\]" checks="." />
-    <suppress files="[/\\].*[/\\]generated[/\\]" checks="." />
```
Comment (carterkozak) [inline .baseline/checkstyle/checkstyle-suppressions.xml:26]: For next time, running `./gradlew baselineUpdateConfig` with updated gradle-baseline plugin should generate these checkstyle config changes automatically

**Review by carterkozak — APPROVED** [review]
lgtm, thanks!

**Comment by carterkozak** [comment]
I think we need to copy the `.policy.yml` config as well

File: .baseline/checkstyle/checkstyle-suppressions.xml (hunk 1 above)
Comment (glynternet) [reply .baseline/checkstyle/checkstyle-suppressions.xml:26]: Ah, awesome. Thanks :)

**Pull request merged**